File: app/src/main/java/com/settlex/android/ui/dashboard/fragment/home/activity/TransactionStatusActivity.java
package com.settlex.android.ui.dashboard.fragment.home.activity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.airbnb.lottie.RenderMode;
import com.settlex.android.R;
import com.settlex.android.data.enums.TransactionStatus;
import com.settlex.android.databinding.ActivityTransactionStatusBinding;
import com.settlex.android.ui.dashboard.DashboardActivity;
import com.settlex.android.util.ui.StatusBarUtil;

import java.util.Objects;

import dagger.hilt.android.AndroidEntryPoint;

/**
 * This screen displays the outcome of a transaction (Pending, Success, Failed).
 */
@AndroidEntryPoint
public class TransactionStatusActivity extends AppCompatActivity {
    private static final String TAG = TransactionStatus.class.getSimpleName();
    private ActivityTransactionStatusBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityTransactionStatusBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setupUiActions();
    }

    private void setupUiActions() {
        StatusBarUtil.setStatusBarColor(this, R.color.white);

        showTransactionData();

        binding.btnDone.setOnClickListener(v -> {
            startActivity(new Intent(this, DashboardActivity.class));
            finishAffinity();
        });
    }

    private void showTransactionData() {
        String txnAmount = getIntent().getStringExtra("amount");
        String txnStatusString = getIntent().getStringExtra("status");

        if (txnAmount == null || txnStatusString == null) {
            Log.e(TAG, "Missing transaction data: amount or status is null");
            return;
        }

        // Display amount
        binding.txnAmount.setText(txnAmount);

        // Convert back to enum
        TransactionStatus txnStatus = null;
        try {
            txnStatus = TransactionStatus.valueOf(txnStatusString);
        } catch (IllegalArgumentException e) {
            Log.e(TAG, "Invalid transaction status: " + txnStatusString, e);
        }

        // Handle UI state based on status
        switch (Objects.requireNonNull(txnStatus)) {
            case SUCCESS -> showSuccessState();
            case PENDING -> showPendingState();
            case FAILED -> showFailedState();
        }
    }

    private void showPendingState() {
        String PENDING_STATUS = "Transaction Pending";
        binding.txnPendingAnim.setVisibility(View.VISIBLE);
        binding.txnPendingAnim.setRenderMode(RenderMode.SOFTWARE);
        binding.txnPendingAnim.playAnimation();
        binding.txnStatus.setText(PENDING_STATUS);
    }

    private void showSuccessState() {
        String SUCCESS_STATUS = "Transaction Successful";
        binding.txnSuccessAnim.setVisibility(View.VISIBLE);
        binding.txnSuccessAnim.setRenderMode(RenderMode.SOFTWARE);
        binding.txnSuccessAnim.playAnimation();
        binding.txnStatus.setText(SUCCESS_STATUS);
    }

    private void showFailedState() {
        String FAILED_STATUS = "Transaction Failed";
        binding.txnFailedAnim.setVisibility(View.VISIBLE);
        binding.txnFailedAnim.setRenderMode(RenderMode.SOFTWARE);
        binding.txnFailedAnim.playAnimation();
        binding.txnStatus.setText(FAILED_STATUS);
    }
}

File: app/src/main/java/com/settlex/android/presentation/transactions/TransactionStatusActivity.kt
package com.settlex.android.presentation.transactions

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.airbnb.lottie.RenderMode
import com.settlex.android.R
import com.settlex.android.data.enums.TransactionStatus
import com.settlex.android.databinding.ActivityTransactionStatusBinding
import com.settlex.android.presentation.common.extensions.getParcelableExtraCompat
import com.settlex.android.presentation.common.extensions.show
import com.settlex.android.presentation.common.extensions.toNairaString
import com.settlex.android.presentation.dashboard.DashboardActivity
import com.settlex.android.presentation.transactions.model.TransactionResult
import com.settlex.android.util.ui.StatusBar
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class TransactionStatusActivity : AppCompatActivity() {
    private lateinit var binding: ActivityTransactionStatusBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityTransactionStatusBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val result = intent.getParcelableExtraCompat<TransactionResult>("transaction_result")
        initViews(result)
    }

    private fun initViews(result: TransactionResult) {
        StatusBar.setColor(this, R.color.colorSurfaceVariant)
        showTransactionData(result)
        setupDoneButton()
    }

    private fun showTransactionData(result: TransactionResult) = with(binding) {
        tvTxnAmount.text = result.amount.toNairaString()
        tvTxnStatus.text = result.message

        when (result.status) {
            TransactionStatus.SUCCESS -> showSuccessState()
            TransactionStatus.PENDING -> showPendingState()
            TransactionStatus.FAILED -> showFailedState(result.errorMessage)
            else -> Unit
        }
    }

    private fun showPendingState() = with(binding) {
        animTxnPending.apply {
            show()
            renderMode = RenderMode.SOFTWARE
        }
    }

    private fun showSuccessState() = with(binding) {
        animTxnSuccess.apply {
            show()
            renderMode = RenderMode.SOFTWARE
        }
    }

    private fun showFailedState(errorMessage: String?) = with(binding) {
        animTxnFailed.apply {
            show()
            renderMode = RenderMode.SOFTWARE
        }

        if (!errorMessage.isNullOrEmpty()) {
            tvError.text = errorMessage
            tvError.show()
        }
    }

    private fun setupDoneButton() = with(binding) {
        tvDone.setOnClickListener {
            navigateToDashboard()
        }
    }

    private fun navigateToDashboard() {
        startActivity(Intent(this, DashboardActivity::class.java))
        finishAffinity()
    }
}
